sdks/java/harness/src/main/java/org/apache/beam/fn/harness/FnHarness.java

```diff
@@ -150,7 +152,11 @@ public static void main(
 
       RegisterHandler fnApiRegistry = new RegisterHandler();
       BeamFnDataGrpcClient beamFnDataMultiplexer =
-          new BeamFnDataGrpcClient(options, channelFactory::forDescriptor, outboundObserverFactory);
+          new BeamFnDataGrpcClient(
+              options,
+              channelFactory.withInterceptors(ImmutableList.of(AddHarnessIdInterceptor.create(id)))
```

Comment:
We should set channelFactory.withInterceptors earlier and share it across all our services.

This will allow us to clean up BeamFnControlClient and stop passing in the id as the channelFactory will already have it.

Reply:
Makes sense.